ui/apps/platform/src/Components/ColumnManagementButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+
+import { ColumnManagementModal } from '@patternfly/react-component-groups';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+
+import { ColumnConfig, ManagedColumns } from 'hooks/useManagedColumns';
+
+export type ColumnManagementButtonProps<ColumnKey extends string> = {
+    managedColumnState: ManagedColumns<ColumnKey>;
+};
+
+function ColumnManagementButton<ColumnKey extends string>({
+    managedColumnState,
+}: ColumnManagementButtonProps<ColumnKey>) {
+    const [isOpen, setOpen] = useState(false);
+    const { columns, setColumns } = managedColumnState;
+    const enabledColumnCount = Object.values<ColumnConfig>(columns).filter(
+        ({ isShown }) => isShown
+    ).length;
+
+    return (
+        <>
+            <ColumnManagementModal
+                appliedColumns={Object.values(columns)}
+                applyColumns={(newColumns) => {
+                    const nextState = Object.fromEntries(
+                        newColumns.map(({ key, isShown }) => [key, isShown ?? false])
+                    );
+                    setColumns(nextState);
+                }}
+                isOpen={isOpen}
+                onClose={() => setOpen(false)}
+            />
+            <Button
+                onClick={() => setOpen(true)}
+                variant={ButtonVariant.secondary}
```

Comment:
Nuance of PatternFly for awareness. I trust your judgement which alternative.
* `ButtonVariant` and its family look like support for application specific wrapper classes:
    https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/Button/Button.tsx#L8-L18
* `variant` property has inline TypeScript string enumeration as source of truth:
    https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/Button/Button.tsx#L79